### PR TITLE
Resolve critical Brakeman issues, bundle update and adding .codeclimate.yml

### PR DESCRIPTION
**Pull request by Brit200313**
Resolves:

- Unsafe reflection method constantize called with parameter value in  app/controllers/admin/pages_controller.rb

https://github.com/pgharts/trusty-cms/issues/154

- Render path contains parameter value in app/controllers/admin/references_controller.rb 

https://github.com/pgharts/trusty-cms/issues/153

This moves TrustyCMS, once deployed, to 2.0.17.

### CODE REVIEW

**Comment by Brit200313** [comment]
@sipple I'm confirming here that this branch now passes TravisCI and CodeClimate. Thanks for reviewing!

File: app/controllers/admin/pages_controller.rb [hunk 1]
```diff
@@ -89,4 +89,12 @@ def initialize_meta_rows_and_buttons
       @meta << {:field => "slug", :type => "text_field", :args => [{:class => 'textbox', :maxlength => 100}]}
       @meta << {:field => "breadcrumb", :type => "text_field", :args => [{:class => 'textbox', :maxlength => 160}]}
     end
+
+  def verify_page_class(page_class)
+    if page_class.constantize.class == Page
```
Comment (sipple) [inline app/controllers/admin/pages_controller.rb]: Does this work for pages that inherit from Page, like `VenuePage` and the like?

File: trusty_cms.gemspec [hunk 2]
```diff
@@ -47,4 +47,5 @@ a general purpose content managment system--not merely a blogging engine.}
   s.add_dependency 'execjs',          '~> 2.2'
   s.add_dependency 'therubyracer',    '~> 0.12'
   s.add_dependency "ckeditor",        "~> 4.1"
+  s.add_dependency "rake",            "< 11.0"
```
Comment (sipple) [inline trusty_cms.gemspec]: Ahhh, broken new version of Rake, you are so delightful.

File: app/controllers/admin/pages_controller.rb (hunk 1 above)
Comment (Brit200313) [inline app/controllers/admin/pages_controller.rb]: Yup! I tested it locally on Convoy. It is really doing the verification twice now to keep away harmful params of trying to become the User class.

File: trusty_cms.gemspec (hunk 2 above)
Comment (Brit200313) [inline trusty_cms.gemspec]: I was pretty pissed. This just popped up in the last week.

Comment (sipple) [inline trusty_cms.gemspec]: Burned us, too. Ugh.

File: app/controllers/admin/pages_controller.rb (hunk 1 above)
Comment (sipple) [inline app/controllers/admin/pages_controller.rb]: Solid!

**Comment by sipple** [comment]
:shipit:

**Pull request merged**